fix some MilitaryAI calculations

Dilvish-fo opened the pull request:
There are still a few places where the MilitaryAI has been using simple addition and subtraction for assessing fleet rating combinations and differences, even though we have had specialized (more accurate) calculations for those in the CombatRatingsAI for some time now.

This PR cleans up those lingering inaccurate uses of plain addition and subtraction.

@Morlic-fo in addition to your substantive review could you give your assessment of suitability to include in 0.4.8?

Dilvish-fo committed:
Add a CombatRatingsAI rating difference helper function

- repurposes/generalizes the calculation from rating_needed()
- provides tooltips that are more informative for this new purpose

Dilvish-fo committed:
Have MilitaryAI always use CombatRatingsAI for assessing ratings

- previously had sometimes used simply addition and subtraction, which are known to be
  inaccurate for these purposes.

Morlic-fo reviewed (dismissed):
For emphasis, see previous discussion

Dilvish-fo committed:
Clarify docstring

- and add extra blank line required for code style

Dilvish-fo committed:
Ensure MilitaryAI doesn't try to assign more forces than are remaining

- "Ensure" is probably a bit overstrong, since the whole running tally of
   what is remaining is remaining is really just an estimate, but that is fine.

Morlic-fo reviewed (changes requested):
> in two out of the three cases

Right, the other case was bugged though as the loop break condition would have never been met. It is a difference if you once allocate a little more than available or do it 100 times more.

Anyway, still one potential bug introduced by your code.

Dilvish-fo committed:
Add ratings magnitude safety check
